src/jq.d.ts

```diff
@@ -0,0 +1,11 @@
+import * as childProcess from "child_process";
+import { PartialOptions } from "./options"
+
+interface run {
+    (filter: string, json: any, options?: PartialOptions, jqPath?: string, spawnOptions?: childProcess.SpawnOptions): Promise<object | string>
+}
+export class JQ {
+    constructor(jqPath?: string, spawnOptions?: childProcess.SpawnOptions);
+    run: run;
+}
+export const run: run;
```

Comment:
I'm not in favour of having a class here, don't see any benefit. Don't share any internal state and in favour of not changing the API neither, I would keep the `run` method working as before, without the `spwanOptions` and `jqPath` arguments. 

They can be really good additions, but I would keep aside.

Reply:
Part of the inspiration for allowing jqPath to be passed at call-time was in situations where node-jq is bundled (ex: webpack) and thus the environment variables are static and changing JQ_PATH during run time has no effect. The spawnOptions is primarily for allowing control over the current working directory (cwd) of the command though following less-is-more philosophy can be just be changed to a cwd option instead of a spawn options bucket.